go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -184,18 +341,97 @@ func (npw *nodePortWatcher) SyncServices(services []interface{}) {
 				serviceInterface)
 			continue
 		}
-		npw.updateServiceFlowCache(service, true)
+		npw.updateServiceFlowCache(service, true, false)
 	}
 
 	npw.ofm.requestFlowSync()
-	syncSharedGatewayIptRules(services)
+	syncSharedGatewayIptRules(services, false, npw.nodeIPs)
+}
+
+// Add new Local Traffic flows if endpoints are local
+func (npw *nodePortWatcher) AddEndpoints(ep *kapi.Endpoints) {
+	name := ktypes.NamespacedName{Namespace: ep.Namespace, Name: ep.Name}
+	// Do nothing if service dosn't exist yet
+	if _, exists := npw.services[name]; exists {
+		// If ETP isn't Local Do nothing
+		if util.ServiceExternalTrafficPolicyLocal(npw.services[name]) {
+			// Only update flows if node local ep state changed from false -> true
+			// Don't care about true -> true or false -> false since correct flows should already be in place
```

Comment:
Right the openflow controller is easy to handle, the problem is we don't have a controller for iptables rules implemented, so I want to avoid churn there

Reply:
Added a simple Add then Delete EP if addresses changed